l3kernel/l3luatex.dtx

```diff
@@ -713,14 +713,18 @@ if luatexbase then
         lua.bytecode[register] = assert(load(str .. "}"))
       end
     end, "ltx.luadata")
-  else
+  end
 %    \end{macrocode}
 %
 % \begin{macro}[int]{get_luadata}
 % \texttt{get_luadata} is only available if data should be restored.
 % It accept the identifier which was used when the data object was registered and
 % returns the associated object. Every object can only be retrieved once.
+% Note that it is possible for both \texttt{register_luadata} and \texttt{get_luadata}
+% to be available, for example while compiling a precompiled preamble.
+% In such a case, data must be registered again to be kept in the next run.
 %    \begin{macrocode}
+  do
```

Comment:
The comment needs to be adjusted since `get_luadata` is afterward always available. Also the `do` / `end` wrapper doesn't seem useful and should be removed.

Reply:
you mean `register_luadata` is the one always available? `get_luadata` should not be available on first run.

The `do`/`end` is because I try to preserve the semantic of the existing code where `local luadata` is a block-local variable.